analyzer/evmabibackfill/evm_abi_backfill.go

```diff
@@ -0,0 +1,214 @@
+	p := &processor{
+		runtime,
+		target,
+		logger,
+	}
+	return item.NewAnalyzer[*abiEncodedItem](
+		evmAbiAnalyzerPrefix+string(runtime),
+		cfg,
+		p,
+		target,
+		logger,
+	)
+}
+
+func (p *processor) GetItems(ctx context.Context, limit uint64) ([]*abiEncodedItem, error) {
+	// There are two types of data we process using a contract abi: transactions and events.
+	// Within a transaction, we process the call data and the revert reason. Since they are
+	// colocated in the same table we can fetch them using a single query.
+	var items []*abiEncodedItem
+	txRows, err := p.target.Query(ctx, queries.RuntimeEvmVerifiedContractTxs, p.runtime, limit)
+	if err != nil {
+		return nil, fmt.Errorf("querying verified contract txs: %w", err)
+	}
+	defer txRows.Close()
+	for txRows.Next() {
+		var tx abiEncodedTx
+		var item abiEncodedItem
+		item.Tx = &tx
+		if err = txRows.Scan(
+			&item.ContractAddr,
+			&item.Abi,
+			&tx.TxHash,
+			&tx.TxData,
+			&tx.TxRevertReason,
+		); err != nil {
+			return nil, fmt.Errorf("scanning verified contract tx: %w", err)
+		}
+		items = append(items, &abiEncodedItem{Tx: &tx})
+	}
+	// Short circuit.
+	if len(items) == int(limit) {
+		return items, nil
+	}
+	eventRows, err := p.target.Query(ctx, queries.RuntimeEvmVerifiedContractEvents, p.runtime, int(limit)-len(items))
+	if err != nil {
+		return nil, fmt.Errorf("querying verified contract evs: %w", err)
+	}
+	defer eventRows.Close()
+	for eventRows.Next() {
+		var ev abiEncodedEvent
+		var item abiEncodedItem
+		item.Event = &ev
+		if err = eventRows.Scan(
+			&item.ContractAddr,
+			&item.Abi,
+			&ev.Round,
+			&ev.TxIndex,
+			&ev.EventBody,
+		); err != nil {
+			return nil, fmt.Errorf("scanning verified contract event: %w", err)
+		}
+		items = append(items, &abiEncodedItem{Event: &ev})
+	}
+	return items, nil
+}
+
+func (p *processor) ProcessItem(ctx context.Context, batch *storage.QueryBatch, item *abiEncodedItem) error {
+	// Unmarshal abi
+	contractAbi, err := abi.JSON(bytes.NewReader(item.Abi))
+	if err != nil {
+		return fmt.Errorf("error unmarshalling abi: %w", err)
+	}
+	// Parse data
+	if item.Event != nil {
+		abiEvent, abiEventArgs, err := abiparse.ParseEvent(item.Event.EventBody.Topics, item.Event.EventBody.Data, &contractAbi)
+		if err != nil {
+			return fmt.Errorf("error processing event using abi for contract %s: %w", item.ContractAddr, err)
+		}
+		eventArgs, err := marshalArgs(abiEvent.Inputs, abiEventArgs)
+		if err != nil {
+			return fmt.Errorf("error processing event args for contract %s: %w", item.ContractAddr, err)
+		}
+
+		batch.Queue(
+			queries.RuntimeEventUpdate,
+			p.runtime,
+			item.Event.Round,
+			item.Event.TxIndex,
+			abiEvent.RawName,
+			eventArgs,
+			abiEvent.Sig,
+		)
+	} else if item.Tx != nil {
+		method, args, err := abiparse.ParseData(item.Tx.TxData, &contractAbi)
+		if err != nil {
+			return fmt.Errorf("error processing tx using abi for contract %s: %w", item.ContractAddr, err)
```

Comment:
This and other error handling in this func: If ABI processing fails, it's almost certainly because of uncompatible data+ABI, which is the tx's fault. We should not fail the item (which in turn will fail the batch), we should mark the item as processed in the DB and just not populate the parsed fields.

Reply:
oh good catch; fixed!